app/controllers/api/base_controller/authentication.rb

```diff
@@ -139,8 +138,8 @@ def validate_system_token_timestamp(timestamp)
 
       def valid_ui_session?
         [
-          valid_authenticity_token?(session, request.x_csrf_token), # CSRF token be set and valid
-          session[:userid].present?,                                # session has a userid stored
+          valid_authenticity_token?(session, request.x_csrf_token) || request.method == 'GET', # CSRF token be set and valid (or GET)
```

Comment:
Is there a way to make a GET request but pass an action so it performs a mutable operation?

Reply:
No. Action is only allowed for post.

Reply:
There _might_ be GETs with side effects, but in general, I think we've tried to avoid that.  If we were to find one, we could always mark that one as needing CSRF protections.

@himdel Would it make sense to move your `request.method == 'GET'` into a method named `request_allowed?` or something?  This way, if we need to deny a particular get request we have a method where we can encapsulate that?